### ISSUE TITLE
Giving "should be" errors when launched

### ISSUE THREAD

**Issue by baturinsky** [fal-works/hlc-compiler#1]
When I try to compile something with `haxelib run hlc-compiler` it gives me 

```
C:\HaxeToolkit\haxe\lib\greeter/0,1,0/src/greeter/Parser.hx:42: characters 3-42 : Array<greeter.CommandArgument> should be greeter.CommandArgumentList
C:\HaxeToolkit\haxe\lib\locator/0,4,0/src/locator/DirectoryPath.hx:182: characters 4-64 : locator.DirectoryPath should be locator.PathString
C:\HaxeToolkit\haxe\lib\hlc-compiler/0,2,0/src/hlc_compiler/GccCommandBuilder.hx:34: characters 3-40 : Array<greeter.CommandArgument> should be greeter.CommandArgumentList
C:\HaxeToolkit\haxe\lib\hlc-compiler/0,2,0/src/hlc_compiler/Main.hx:141: characters 47-49 : Array<locator.FileOrDirectoryRef> should be locator.FileOrDirectoryList
C:\HaxeToolkit\haxe\lib\locator/0,4,0/src/locator/FileOrDirectoryList.hx:80: characters 3-43 : Array<locator.FileOrDirectoryRef> should be locator.FileOrDirectoryList
C:\HaxeToolkit\haxe\lib\locator/0,4,0/src/locator/DirectoryRef.hx:111: lines 111-112 : Array<locator.FileOrDirectoryRef> should be locator.FileOrDirectoryList
C:\HaxeToolkit\haxe\lib\greeter/0,1,0/src/greeter/CommandLine.hx:61: characters 23-32 : Array access is not allowed on greeter.CommandArgumentList
C:\HaxeToolkit\haxe\lib\locator/0,4,0/src/locator/FileList.hx:42: characters 3-33 : Array<locator.FileRef> should be locator.FileList
C:\HaxeToolkit\haxe\lib\locator/0,4,0/src/locator/DirectoryList.hx:35: characters 3-44 : Array<locator.DirectoryRef> should be locator.DirectoryList
```

**Comment by tobil4sk** [comment]
Found the cause of the issue, it's a bug with the Haxe compiler and not this specific library:

https://github.com/HaxeFoundation/haxe/issues/9169

Haven't tested it, but using 4.1.5 might be a way of getting around this issue for now without having to modify code due to a compiler bug.

**Comment by tobil4sk** [comment]
In the end, it turns out this is not a bug, and isn't the same thing mentioned in the issue I linked in the last comment. These failures are the result of an intended change which is documented here: https://github.com/HaxeFoundation/haxe/pull/9698.

The following PRs to the dependencies of this project should address this issue:
- https://github.com/fal-works/sinker/pull/1
- https://github.com/fal-works/locator/pull/1

**Comment by fal-works** [comment]
Hi, thank you so much for the PRs and I'm sorry for leaving this issue for such a long time!
I was just about to update the depending libraries. So I'll check the PRs and probably include them in the next versions.
(I'm also going to change the license from MIT to Apache 2.0, if you don't mind; both are quite permissive)

**Comment by tobil4sk** [comment]
Thanks for the quick reply! Glad to hear the news, this tool is a real life saver when it comes to compiling hashlink/c, so I and many others will be very grateful for a release! :) btw, both changes are one line each, so it shouldn't be too great a hassle to review. Regarding the license change, I have no issue with that.

You might also want to consider having this library run on neko bytecode. The way you do this is by compiling the tool with the neko target to make a "run.n" file (this file is usually committed to source control to make it easier for library users, although it has to be updated manually). Despite the deprecation of neko this still has some advantages, including the fact that there is no compilation step, which avoids potential compiler errors like this in the future. It also makes the tool slightly more responsive on the CLI, but you can evaluate this for yourself (though if you'd like help setting this up, please let me know!).

**Comment by fal-works** [comment]
After fixing issues in the dependencies, I finally managed to confirm that `hlc-compiler` works in the recent environment (at least on my computers).

Also updated:
- this library itself (<https://lib.haxe.org/p/hlc-compiler/>) and
- its example project (<https://github.com/fal-works/hlc-compiler-sample>).

By the way the idea of neko bytecode sounds good too (thanks again), so I've also included a "run.n" in the recent version.
Hope this works!
